lib/connection/process.coffee

```diff
@@ -36,28 +38,49 @@ module.exports = jlprocess =
 
   script: (s...) -> @packageDir 'script', s...
 
+  # spawn Julia in the first open project folder (or its parent folder for files)
   workingDir: ->
     paths = atom.workspace.project.getDirectories()
-    if paths.length == 1 and fs.statSync(paths[0].path).isDirectory()
-      paths[0].path
+    if fs.statSync(paths[0].path).isFile()
+      path.dirname paths[0].path
     else
-      process.env.HOME || process.env.USERPROFILE
+      paths[0].path
 
   jlpath: ->
     p = atom.config.get("julia-client.juliaPath")
     if p == '[bundle]' then p = @bundledExe()
     p
 
-  checkPath: (path) ->
-    new Promise (resolve) ->
-      fs.exists path, (exists) ->
+  checkPath: (p) ->
+    new Promise (resolve, reject) =>
+      # check whether path exists
+      fs.exists p, (exists) =>
```

Comment:
We might be able to flatten this big chain a bit / reduce the filesystem calls by using `fs.stat` to both check for existence and file-ness.